src/AppM.purs

```diff
@@ -181,29 +183,34 @@ withUser
    . MonadAff m 
   => LogMessages m 
   => Navigate m 
+  => MonadAsk Env m
   => Authenticate m
   => (Username -> Json -> Either String a)
-  -> Request Json
+  -> (URL -> Request Json)
   -> m (Either String a)
 withUser decode req =
```

Comment:
I think it's worth creating an `Api.Utils` module that just has these helper functions in it, rather than have them in `AppM`. They're creating quite a bit of clutter down the bottom of the module.